### PR TITLE
Fixes for VirtualBox

### LINKED ISSUE #742: VM: Install python packages fails on uninstall pexpect
Trying to create a new VM.

For whatever reason
https://github.com/SyneRBI/SIRF-SuperBuild/blob/ec4b9c50bb650101e232590405c06eaf417256a6/VirtualBox/scripts/INSTALL_python_packages.sh#L70
triggers an uninstall of `pexpect` with `pip`, but this fails as there's one installed by `apt` I guess.

screenshot (sorry, no copy-paste capability on this VM yet)
 image

The installed package is at `/usr/lib/python3/dist-packages/pexpect`. Maybe it got there via https://github.com/SyneRBI/SIRF-SuperBuild/blob/d9de45ba5163eb175d6bb8cd37a2031755fd4d26/VirtualBox/scripts/INSTALL_prerequisites_with_apt-get.sh#L39-L40

### PR DESCRIPTION
**Pull request by paskino**
Creates the VM from the docker files.

Installs custom pip. 

closes #742

### CODE REVIEW

File: VirtualBox/scripts/INSTALL_prerequisites_with_apt-get.sh [hunk 1]
```diff
@@ -36,11 +39,11 @@ ${APT_GET_INSTALL} expect
...
 # some extra package needed for jupyter
-qt=pyqt5
-${APT_GET_INSTALL} python3-dev python3-pip python3-tk python3-${qt} python3-${qt}.qtsvg python3-${qt}.qtwebkit
+# qt=pyqt5
+# ${APT_GET_INSTALL} python3-dev python3-pip python3-tk python3-${qt} python3-${qt}.qtsvg python3-${qt}.qtwebkit
 
-echo "Installing CIL pre-requisites..."
-${APT_GET_INSTALL} cython3 python3-h5py python3-wget
+# echo "Installing CIL pre-requisites..."
+# ${APT_GET_INSTALL} cython3 python3-h5py python3-wget
```
Comment (KrisThielemans) [inline VirtualBox/scripts/INSTALL_prerequisites_with_apt-get.sh:46]: let's get rid of all these

File: VirtualBox/scripts/INSTALL_python_packages.sh [hunk 2]
```diff
@@ -65,12 +65,25 @@ while (( "$#" )); do
...
 done
 
-$PYTHON -m pip install $PIPOPTIONS --upgrade pip wheel setuptools
-$PYTHON -m pip install $PIPOPTIONS --only-binary=numpy,scipy,matplotlib numpy scipy matplotlib nose coverage docopt deprecation nibabel pytest tqdm
-$PYTHON -m pip install $PIPOPTIONS jupyter spyder
-$PYTHON -m pip uninstall $PIPOPTIONS -y spyder-kernels
+export DEBIAN_FRONTEND=noninteractive
+SUDO=sudo
+APT_GET_INSTALL="$SUDO apt-get install -y --no-install-recommends"
+${APT_GET_INSTALL} ${PYTHON}-dev
```
Comment (KrisThielemans) [inline VirtualBox/scripts/INSTALL_python_packages.sh:71]: move to `INSTALL_...with_apt...`

File: VirtualBox/scripts/INSTALL_python_packages.sh [hunk 3]
```diff
@@ -65,12 +65,25 @@ while (( "$#" )); do
...
+${PYTHON} get-pip.py
+rm get-pip.py
+
+${PYTHON} -m pip install $PIPOPTIONS -U setuptools wheel
+${PYTHON} -m pip install $PIPOPTIONS -U -r ${SCRIPTS}/requirements.txt
+${PYTHON} -m pip install $PIPOPTIONS -U -r ${SCRIPTS}/requirements-service.txt
+# $PYTHON -m pip install $PIPOPTIONS --upgrade pip wheel setuptools
+# $PYTHON -m pip install $PIPOPTIONS --only-binary=numpy,scipy,matplotlib numpy scipy matplotlib nose coverage docopt deprecation nibabel pytest tqdm
+# $PYTHON -m pip install $PIPOPTIONS jupyter spyder
+# $PYTHON -m pip uninstall $PIPOPTIONS -y spyder-kernels
```
Comment (KrisThielemans) [inline VirtualBox/scripts/INSTALL_python_packages.sh:85]:
```suggestion
```

File: VirtualBox/scripts/UPDATE.sh [hunk 4]
```diff
@@ -141,9 +141,21 @@ if [ -d $SIRF_SRC_PATH/CCPPETMR_VM ]; then
...
-# best to use full path
+
+# best to use full path for python3/cython
+PYTHON_EXECUTABLE=$(which python3)
+if which python3; then
+  PYTHON_EXECUTABLE=$(which python3)
+else
+  PYTHON_EXECUTABLE=$(which python)
+fi
 PYTHON_EXECUTABLE=/usr/bin/python3
```
Comment (KrisThielemans) [inline VirtualBox/scripts/UPDATE.sh:152]:
```suggestion
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Comment by KrisThielemans** [comment]
so, did it work?

**Comment by KrisThielemans** [comment]
doing most of these myself at the moment

**Comment by KrisThielemans** [comment]
I'm going to squash-merge this now, and then fix some things later.

**Comment by paskino** [comment]
> so, did it work?

Yes, it did! I opened the PR only after it all worked!

**Comment by KrisThielemans** [comment]
great. I'm changing it anyway 😄 PR coming later today.

**Pull request merged**